### ISSUE TITLE
Proton 5.13-2 RC testing

### ISSUE THREAD

**Issue by aeikum** [ValveSoftware/Proton#4360]
We've put together a new 5.13 build for you to test.

In the Steam client, the Proton 5.13 app should have a "next" beta branch which you can choose to start testing the 5.13-2 release candidates (**note that the name of the build in the Steam Settings dialog will not be updated**). I will post changes here when we push new builds. The source for the latest RC build is available on the proton_5.13-next branch in these repositories. This branch may receive forced updates.

In this issue, we are interested only in problems that are new to the 5.13-2 RC builds. If you find new problems in the "next" branch, please confirm that the problem does not occur in on the "default" branch before reporting it here.

Here is the tentative changelog. As always, this changelog has not yet been verified by our QA staff, and can change before the final release as we add or remove features during RC testing.

-Update vkd3d-proton to version 2.0.

-Fix Assetto Corsa Competizione in VR mode.

-Proton log directory can be configured with `PROTON_LOG_DIR`.

-Fix Killer Instinct crash on exit.

-Fix Risk of Rain 2 multiplayer lobbies.

-Fix games not launching in Uplay Connect, and Origin crashing when updating.

-Fix mouse cursor in Mount & Blade II: Bannerlord.

-Fix SpellForce crash on launch on some systems.

-Fix very long load times in Warhammer 40k: Inquisitor.

-Restore behavior of horizontal scrolling input from previous Proton versions.

**Comment by l33tlinuxh4x0r** [comment]
Controller hot plugging is still broken in 5.13-2

**Comment by aeikum** [comment]
Hello. I've just uploaded a new RC build. This new version has the following changes:

-Risk of Rain 2 multiplayer lobbies should actually be working now.

-`PROTON_LOG_DIR` set in `user_settings.py` should be respected now.

-Atelier Ryza should be playable again.

-Incorrect colors in some videos should be fixed.

**Comment by ranplayer** [comment]
I've just tested Killer Instinct with a fresh prefix. The only issue I've found so far: after a fresh install if you try to access "Help & Options" (Single Player) the interface hangs and you have to manually close the game window. When you launch the game again it works normally. Now the game is not crashing anymore after a fight ends.

**Comment by ranplayer** [comment]
My setup:
GPU: GTX 970
Driver : 455.38
O.S: ArchLinux
DE: KDE

**Comment by niobium93** [comment]
Call of Duty: WWII (476600) now works! Though it seems a little confused about display drivers:
![2020-11-10-18:56:42-screenshot](https://user-images.githubusercontent.com/1523948/98718982-d37ab400-2397-11eb-8e98-113714df6d5b.png)
After dismissing that warning (which shows up every time I start the game) the game itself works fine, besides the usual shader compilation slowdown. The audio also crackles. The multiplayer (476620) still refuses to launch.
[system-information.txt](https://github.com/ValveSoftware/Proton/files/5519612/system-information.txt)
[steam-476600.log.gz](https://github.com/ValveSoftware/Proton/files/5519624/steam-476600.log.gz)
[steam-476620.log](https://github.com/ValveSoftware/Proton/files/5519626/steam-476620.log)

**Comment by GloriousEggroll** [comment]
@aeikum The COD:WWII display driver issue is due to old driver being defined in atiadlxx:
```
atiadlxx]$ grep -rni . -e 16.11
./atiadlxx_main.c:141:    "16.11.2",
./atiadlxx_main.c:142:    "16.11.2",
./atiadlxx_main.c:147:    "16.11.2",
./atiadlxx_main.c:148:    "16.11.2",
./atiadlxx_main.c:149:    "16.11.2",
```
Changing these to match AMD's current version 20.11.1 fixes it in my local build.

**Comment by zelfir** [comment]
I still have issues with this release as it seems to not use my nvidia GPU but the Intel one instead despite the fact I'm running on nvidia discrete GPU as stated by nvidia-smi: 
`NVIDIA-SMI 455.28       Driver Version: 455.28       CUDA Version: 11.1`
example in the log here attached with the game Control :
[steam-870780.log](https://github.com/ValveSoftware/Proton/files/5520892/steam-870780.log)

**Comment by StefanTT** [comment]
Tested some games in VR with this release: Assetto Corsa Competitione, BoxVR, Zero Caliber, Project Cars 2.

All work fine except Project Cars 2, it crashes / hangs on startup in VR mode. But as this game currently is broken in VR anyways it is not much of a difference.

This is on Arch Linux with Nvidia 455.38 and linux 5.9.6

**Comment by Leopard1907** [comment]
@zelfir This is due to runtime and there is no fix for it yet.

**Comment by Samsagax** [comment]
Tested both 5.13 versiones and both wont pick up my dGPU on a PRIME configuration. `gamemoderun` is ignored and environment variables too. I'm starting my games with this line: `gamemoderun %command%` and it seems to do nothing. Also starting them with `__NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia __VK_LAYER_NV_optimus=NVIDIA_only %command%` will ignore the PRIME card offload. I think is related to the way Soldier is set up. I'm able to test further if i could get any pointers.

**Comment by kisak-valve** [comment]
Hello @zelfir, @Samsagax, that issue is currently being tracked at https://github.com/ValveSoftware/steam-runtime/issues/295.

**Comment by Samsagax** [comment]
Ok. After a workaround for [this](https://github.com/ValveSoftware/Proton/issues/4360#issuecomment-725407497) AOE2: DE works with no performance issues but AOE2: HD doesn't as [this issue](https://github.com/ValveSoftware/Proton/issues/72#issuecomment-709831719) is still present.

**Comment by jotomo** [comment]
I have two audio devices (soundcard, BT headphones), where the latter is configured as the active device (KDE, Pulseaudio on Arch Linux). With Proton 5.0 and Dirt Rally 2.0, the audio is played through the headphones, whereas with 5.13-RC1/2, the audio is played via soundcard (first audio device). Disabling the sound card on the pulseaudio level doesn't help. Not sure where or at what layer to look for possible causes, pointers appreciated.

**Comment by StefanTT** [comment]
@jotomo as a workaround you can use pavucontrol to change the output device of a running application.

**Comment by Emanem** [comment]
Vulkan Overlays (based off MESA overlay) don't seem to be working - anyone has an idea already why that is?

**Comment by jotomo** [comment]
> @jotomo as a workaround you can use pavucontrol to change the output device of a running application.

Thanks! Interestingly, with 5.0 pavucontrol shows _ALSA plugin-in [wine64-preloader]: ALSA playback_ while with 5.13 this stream doesn't appear.
I've blacklisted the modules for the PCI soundcard, so only the BT headphones remain which still works for 5.0[1], but with 5.13, Dirt Rally 2.0 hangs on start[2]. Loading the modules for the PCI soundcard manually, so that card becomes the second audio device has no effect, audio still plays via PCI soundcard, with no stream showing in pavucontrol[3].

Edit: the issue isn't universal, e.g. Lonely Mountains Downhill works correctly with 5.13 (no change from 5.0).

[1] [steam-690790-proton5.10-10-game-starts-and-audio-plays-with-only-bt-audio-available.log](https://github.com/ValveSoftware/Proton/files/5540554/steam-690790-proton5.10-10-game-starts-and-audio-plays-with-only-bt-audio-available.log)
[2] [steam-690790-proton5.13rc2-hangs-with-only-bt-audio-available.zip](https://github.com/ValveSoftware/Proton/files/5540547/steam-690790-proton5.13rc2-hangs-with-only-bt-audio-available.zip) (137 MB uncompressed)
[3] [steam-690790-proton5.13rc2-bt-audio-as-first-device-still-plays-via-pci-soundcard.log](https://github.com/ValveSoftware/Proton/files/5540550/steam-690790-proton5.13rc2-bt-audio-as-first-device-still-plays-via-pci-soundcard.log)

**Comment by StefanTT** [comment]
Interesting. I am also on Arch and for me pavucontrol shows the stream as "dirtrally2.exe: Simple direct media layer"
with the latest Proton 5.13 beta.

**Comment by AirstrikeIvanov** [comment]
I also appear to have the same issue with multiple audio devices and 5.13-2 picking the wrong one. Reverting to 5.0-10 fixes the issue temporarily.

**Comment by aeikum** [comment]
Thanks all for testing. 5.13-2 went live last Friday.

**Comment by jotomo** [comment]
@AirstrikeIvanov FYI: following along https://github.com/ValveSoftware/steam-runtime/issues/357#issuecomment-768247003 I found the issue (wrong audio device used) on my system: I had set the environment variable `SDL_AUDIODRIVER=alsa` in my _.bashrc_ at some point for DOSBox. Seems Proton <5.13 didn't care, but the new runtime used by 5.13 does take that into account. If you're on Slackware the referenced bug may be hitting you, otherwise maybe check your environment variables. HTH.
